### ISSUE TITLE
Misleading Error Message with == and valueOf

### ISSUE THREAD

**Issue by SteveBeeblebrox** [microsoft/TypeScript#53757]
# Bug Report

### 🔎 Search Terms
always false, valueOf, equality

### 🕗 Version & Regression Information
- This changed between versions 4.7.4 and 4.8.4 (and still happens on nightly)

### ⏯ Playground Link
[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgziA2CmB0w4EMBOAKAlAWAFCkhgWhAHN0ByAMxBAoAIBeR+gbwDdloBXWAeSpY2qWABduqMPWq0KAXzmZ6AemX1RqXkA)

### 💻 Code

```ts
console.clear()
console.log('foo' == {valueOf() {return 'foo'}}) // true
```

### 🙁 Actual behavior
The above code snippet produces an error on the `==` condition: `This condition will always return 'false' since JavaScript compares objects by reference, not value.(2839)` which would lead the user to believe that `console.log(...)` would always print false. **In reality, the line prints true**. The object on the right is coerced to a string using the `valueOf()` method so the comparison becomes `'foo' == 'foo'` which is true, but TypeScript claims otherwise.

### 🙂 Expected behavior
Even if TypeScript cannot determine that this condition would be always true given the usage of `valueOf`, I would expect the error message to say something more like `...may always return false...`
instead of the more certain `...will always return false...` so that the user is not told that something will certainly happen when it may not.

**Comment by MartinJohns** [comment]
I'd call this a duplicate of #2361. `valueOf()` is just not really supported by the type system.

**Comment by SteveBeeblebrox** [comment]
I don't expect TypeScript to understand the `valueOf()`, but it would be nice if the error message wording didn't contradict runtime behavior.

**Comment by fatcerberus** [comment]
What do you propose?  The error message is true if you name the method literally anything else, so you’re de facto asking typescript to understand the semantics of `valueOf`, which it currently doesn’t.

**Comment by RyanCavanaugh** [comment]
It's a waste of time and effort to make the message less clear for the sake of satisfying cases that can only be hit through specific construction.
